### PR TITLE
[Part-1] Refactor k8s related code into a separate module

### PR DESCRIPTION
**Pull request by varshaprasad96**
# Issue link

# What changes have been made

This commit is first of the chain in refactoring the cluster/ package. This moves the auth package into a separate module.

The unit tests have not been refactored based on the discussion in: [here](https://redhat-internal.slack.com/archives/C07KGP1MGLE/p1727348601272359). A separate PR will be created to refactor all the tests. 

# Verification steps

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [X] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: src/codeflare_sdk.egg-info/PKG-INFO [hunk 1]
```diff
@@ -1,4 +1,4 @@
 Metadata-Version: 2.1
-Name: codeflare-sdk
+Name: codeflare_sdk
```
[the remark this answers is not in the document]
Comment (ChristianZaccaria) [reply src/codeflare_sdk.egg-info/PKG-INFO:2]: I'm not 100% sure, but I think the name in the `PKG-INFO` file is the name used in PyPi. In PyPi the sdk package is named `codeflare-sdk`. Perhaps it's best to revert this. - I'll see if I can verify this

[the remark this answers is not in the document]
Comment (Bobbins228) [reply src/codeflare_sdk.egg-info/PKG-INFO:2]: I see, could that be because the project structure is `src/codeflare_sdk`?

**Review by ChristianZaccaria — COMMENTED** [review]
/lgtm great work!

File: src/codeflare_sdk.egg-info/PKG-INFO (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChristianZaccaria) [reply src/codeflare_sdk.egg-info/PKG-INFO:2]: The name specified here will be used as the distribution name on PyPi: https://packaging.python.org/en/latest/specifications/core-metadata/#name

[the remark this answers is not in the document]
Comment (Ygnas) [reply src/codeflare_sdk.egg-info/PKG-INFO:2]: https://issues.redhat.com/browse/RHOAIENG-6635 I still think this should be done at some point and egg-info not commited.

[the remark this answers is not in the document]
Comment (Bobbins228) [reply src/codeflare_sdk.egg-info/PKG-INFO:2]: Okay we will have to revert this. Would it be best to just add it to .gitignore?

Comment (Bobbins228) [inline src/codeflare_sdk.egg-info/PKG-INFO:2]:
```suggestion
Name: codeflare-sdk
```
Perhaps if you could add this file to the `.gitignore`?

**Review by Bobbins228 — CHANGES_REQUESTED** [review]

File: src/codeflare_sdk.egg-info/PKG-INFO (hunk 1 above)
Comment (ChristianZaccaria) [reply src/codeflare_sdk.egg-info/PKG-INFO:2]: Perhaps the whole `codeflare_sdk.egg-info` directory should be in the .gitignore. Then we could close: https://issues.redhat.com/browse/RHOAIENG-6635

Comment (Ygnas) [reply src/codeflare_sdk.egg-info/PKG-INFO:2]: Yes it is possible to remove tracked file and add ignore it.

**Comment by Bobbins228** [comment]
I think we are missing imports for `__kube_api_error_handling` in `tests/upgrades/raycluster_sdk_upgrade_sleep_test.py` and `tests/upgrades/raycluster_sdk_upgrade_test.py`

File: src/codeflare_sdk.egg-info/SOURCES.txt [hunk 2]
```diff
@@ -7,17 +7,19 @@ src/codeflare_sdk.egg-info/SOURCES.txt
...
+src/codeflare_sdk/common/kubernetes_cluster/__init__.py
+src/codeflare_sdk/common/kubernetes_cluster/auth.py
+src/codeflare_sdk/common/kubernetes_cluster/kube_api_helpers.py
 src/codeflare_sdk/job/__init__.py
 src/codeflare_sdk/job/ray_jobs.py
 src/codeflare_sdk/utils/__init__.py
 src/codeflare_sdk/utils/demos.py
 src/codeflare_sdk/utils/generate_cert.py
 src/codeflare_sdk/utils/generate_yaml.py
-src/codeflare_sdk/utils/kube_api_helpers.py
```
Comment (ChristianZaccaria) [inline src/codeflare_sdk.egg-info/SOURCES.txt:22]: nit: for this file to be ignored in `.gitignore`, it needs to be deleted.
No need to hold merge for this.

**Review by Ygnas — APPROVED** [review]

**Review by Bobbins228 — COMMENTED** [review]
/lgtm /approve

**Review by ChristianZaccaria — APPROVED** [review]

**Review by Bobbins228 — APPROVED** [review]

**Pull request merged**